app/src/main/kotlin/com/adesso/movee/scene/tvshowdetail/TvShowDetailViewModel.kt

```diff
@@ -46,13 +41,8 @@ class TvShowDetailViewModel @Inject constructor(
 
     fun fetchTvShowCredits(id: Long) {
         if (_tvShowCredits.value == null) {
-            bgScope.launch {
-                val tvShowCreditResult =
-                    fetchTvShowCreditsUseCase.run(FetchTvShowCreditsUseCase.Params(id))
-
-                onUIThread {
-                    tvShowCreditResult.either(::handleFailure, ::postTvShowCredits)
-                }
+            fetchTvShowCreditsUseCase.invoke(viewModelScope, FetchTvShowCreditsUseCase.Params(id)) {
+                it.either(::handleFailure, ::postTvShowCredits)
```

Comment:
postTvShowCredits() uses LiveData's setValue() function which must be called from main thread. Either we need to use postValue(), or switch context when calling postTvShowCredits()